lldb/include/lldb/Host/Socket.h

```diff
@@ -34,9 +34,11 @@ namespace lldb_private {
 #if defined(_WIN32)
 typedef SOCKET NativeSocket;
 typedef lldb::pipe_t shared_fd_t;
+typedef const char *set_socket_option_arg_type;
```

Comment:
Instead of exposing this quirk, it would be better to have a static version of `Socket::SetOption` to hide it.

Reply:
Updated. Fixed also a typo
`set`_socket_option_arg_type option_value_p =
      reinterpret_cast<`get`_socket_option_arg_type>(&option_value);